### ISSUE TITLE
Allow events of foreign contract types to be visible in lookup.

### ISSUE THREAD

**Issue by scnale** [ethereum/solidity#13928]
## Abstract

I found myself needing to reference another contract's events while writing unit tests in the Solidity language:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

contract Test {
    event MyEvent(uint256 index);

    function testingEvents() external returns (bytes32) {
        // Works
        return Test.MyEvent.selector;
    }

    function testingEvents2() external returns (bytes32) {
        // Works
        return MyEvent.selector;
    }
}


contract AnotherTest {
    function testingEvents() external returns (bytes32) {
        // Doesn't compile due to failed lookup of `MyEvent`.
        return Test.MyEvent.selector;
    }
}
```

I'm actually not sure whether to format this as a feature request or a bug :sweat_smile:

## Motivation

In the [Foundry](https://book.getfoundry.sh/) framework, you can write unit tests with Solidity. If I want to filter events emitted in logs, I can hash the signature manually:
```solidity
bytes32 selector = keccak256("MyEvent(uint256)");
```

However, it would be cleaner to actually reference the event so it's harder for the selector definition to go stale:

```solidity
bytes32 selector = Test.MyEvent.selector;
```

## Specification

I don't think this introduces new syntax.

Regarding semantics, I'm personally only interested in being able to evaluate an event's selector.

However, I wouldn't know whether it makes sense to allow emission of foreign events in your own contract:

```solidity
contract Test {
    event MyEvent(uint256 index);
}


contract AnotherTest {
    function testingEvents() external returns (bytes32) {
        emit Test.MyEvent(5);
        return 0;
    }
}
```

Update: Actually, to assert that an event was emitted, [`expectEmit`](https://book.getfoundry.sh/cheatcodes/expect-emit) is used, and it requires emitting the event with the expected arguments. So it does make sense to allow emitting foreign events in the context of test writing at least.

## Backwards Compatibility

As far as I can see, there should be no backwards incompatibility due to the fact that function, modifier, struct and event identifiers that overlap are disallowed. I might be missing something though.

**Comment by cameel** [comment]
This is already planned, but we need to have #13086 first. Which should happen pretty soon (it already has a pending PR).

**Comment by cameel** [comment]
We actually already have another issue covering this, so closing this one to keep discussions in a single place: #9737.

**Comment by cameel** [comment]
Ah wait, they're not the same. Still, the upcoming change will cover both because the main thing preventing us from having events at file level was the problem with referencing foreign events.

**Comment by caffeinum** [comment]
This seems to be fixed in `0.8.25`. The code from the OP compiles without errors
